crates/wasmcloud-host/src/messagebus/latticecache_client.rs

```diff
@@ -0,0 +1,460 @@
+    /// function will return false if any errors happen while communicating with the message
+    /// bus.
+    pub async fn contains_oci_mapping(&self, oci_ref: &str) -> bool {
+        let key = prefix("ocis");
+        let args = SetQueryArgs { key };
+        let inv = self.invocation_for_provider(OP_SET_QUERY, &serialize(&args).unwrap());
+        invoke_as::<SetQueryResponse>(&self.provider, inv)
+            .await
+            .map(|qr| qr.values.contains(&oci_ref.to_string()))
+            .unwrap_or(false)
+    }
+
+    /// Adds an OCI reference mapping to the cache. The OCI reference is an OCI compliant
+    /// URL, while the public key is the public key of either a capability provider or
+    /// an actor
+    pub async fn put_oci_mapping(&self, oci_ref: &str, public_key: &str) -> Result<()> {
+        // KV context: KEY is {prefix}:oci_{oci_ref}, VAL is public key
+        // SET KEY: {prefix}:ocis, SET MEMBERS: oci_ref
+        let key = oci_key(oci_ref);
+        let args = SetArgs {
+            key,
+            value: public_key.to_string(),
+            expires: 0,
+        };
+        let inv = self.invocation_for_provider(OP_SET, &serialize(&args)?);
+        let inv_r = self.provider.send(inv).await?;
+        if let Some(e) = inv_r.error {
+            let s = format!("Failed to put OCI mapping: {}", e);
+            error!("{}", s);
+            return Err(s.into());
+        }
+
+        let key = prefix("ocis");
+        let args = SetAddArgs {
+            key,
+            value: oci_ref.to_string(),
+        };
+        let inv = self.invocation_for_provider(OP_SET_ADD, &serialize(&args)?);
+        let inv_r = self.provider.send(inv).await?;
+        if let Some(e) = inv_r.error {
+            let s = format!("Failed to put OCI mapping(set): {}", e);
+            error!("{}", s);
+            Err(e.into())
+        } else {
+            Ok(())
+        }
+    }
+
+    pub async fn remove_oci_mapping(&self, oci_ref: &str) -> Result<Option<String>> {
+        if let Some(s) = self.lookup_oci_mapping(oci_ref).await? {
+            self.remove_oci(oci_ref).await?;
+            Ok(Some(s))
+        } else {
+            Ok(None)
+        }
+    }
+
+    /// Retrieves the public key (if one exists) corresponding to the supplied OCI image
+    /// reference
+    pub async fn lookup_oci_mapping(&self, oci_ref: &str) -> Result<Option<String>> {
+        let key = oci_key(oci_ref);
+        let args = GetArgs { key };
+        let inv = self.invocation_for_provider(OP_GET, &serialize(&args)?);
+        let gr: GetResponse = invoke_as(&self.provider, inv).await?;
+        if gr.exists {
+            Ok(Some(gr.value))
+        } else {
+            Ok(None)
+        }
+    }
+
+    /// Retrieves the list of all currently known OCI image references. You can this list in a loop to
+    /// retrieve all of the image reference mappings.
+    pub async fn get_oci_references(&self) -> Result<Vec<String>> {
```

Comment:
Does this comment mean to say that I can iterate through this list to retrieve all of the image reference mappings?